spackbot/pr_add_comments.py

```diff
@@ -39,17 +43,139 @@ def say_hello():
 """
 
 
+def get_style_message(output):
+    """
+    Given a terminal output, wrap in a message
+    """
+    return (
+        """
+I was able to run `spack style --fix` for you!
+    
+```bash
+%s
+```    
+Keep in mind that I cannot fix your flake8 or mypy errors, so if you have any you'll need to fix them and update the pull request."""
+        % output
+    )
+
+
+def is_up_to_date(output):
+    """
+    A commit can fail if there are no changes!
+    """
+    return "branch is up to date" in output
+
+
+async def fix_style(event, gh, token):
+    """
+    Respond to a request to fix style.
+
+    We first retrieve metadata about the pull request. If the request comes
+    from anyone with write access to the repository, we commit, and we commit
+    under the identity of the original person that opened the PR.
+    """
+    response = requests.get(event.data["issue"]["pull_request"]["url"])
+    pr = response.json()
+
+    # Get the sender of the PR - do they have write?
+    sender = event.data["sender"]["login"]
+    repository = event.data["repository"]
+    collaborators_url = repository["collaborators_url"]
+
+    # If they don't have write, we don't allow the command
+    if not await found(gh.getitem(collaborators_url, {"collaborator": sender})):
+        logger.info(f"Not found: {sender}")
+        return (
+            "Sorry %s, I cannot do that for you. Only users with write can make this request!"
+            % sender
+        )
+
+    # Tell the user the style fix is going to take a minute or two
+    message = "Let me see if I can fix that for you! This might take a moment..."
+    await gh.post(event.data["issue"]["comments_url"], {}, data={"body": message})
+
+    # Get the username of the original committer
+    user = pr["user"]["login"]
+    email = "%s@users.noreply.github.com" % user
+
+    # Get the branch and pull request url
+    branch = pr["head"]["ref"]
+    clone_url = pr["head"]["repo"]["svn_url"]
+    naked_url = clone_url.replace("https://", "", 1)
+
+    # If for some reason the message below isn't generated?
+    message = "There was a problem running fixes! @spackbot needs a maintainer's help!"
+
+    # At this point, we can clone the repository and make the change
+    with temp_dir() as cwd:
+        git("clone", "-b", branch, clone_url)
+        os.chdir("spack")
+        git("config", "--local", "user.name", user)
+        git("config", "--local", "user.email", email)
+
+        # This will authenticate the push with the app token
+        git(
+            "remote",
+            "set-url",
+            "origin",
+            "https://%s:%s@%s.git" % (user, token, naked_url),
+        )
+
+        # We need to add develop remote for this to work
+        git("remote", "add", "upstream", "https://github.com/spack/spack")
+        git("fetch", "upstream")
+
+        # This won't work if we are already on a develop branch
+        if branch != "develop":
```

Comment:
I think this gave me an error in testing, but now I'm not sure if it was because of not having the spack style command. I don't think we can checkout a second develop branch from upstream given that the PR is from a develop branch (but possibly should test this again).